src/modules/esl-share/core/esl-share-config.ts

```diff
@@ -31,32 +35,137 @@ export interface ESLShareGroupConfig {
-  }, [] as ESLShareButtonConfig[]);
+    return instance;
+  }
+
+  protected _buttons: ESLShareButtonConfig[] = [];
+  protected _groups: ESLShareGroupConfig[] = [];
+
+  /** @returns config of buttons */
+  public get buttons(): ESLShareButtonConfig[] {
+    return this._buttons;
+  }
+
+  /** @returns config of groups */
+  public get groups(): ESLShareGroupConfig[] {
+    return this._groups;
+  }
+
+  /**
+   * Selects the buttons for the given list and returns their configuration.
+   * @returns config of buttons
+   */
+  public getList(list: string): ESLShareButtonConfig[] {
+    return list.split(' ').reduce((res, item) => {
+      const [btnName, groupName] = item.split('group:');
+      if (groupName) {
+        const groupConfig = this.getGroup(groupName);
+        if (groupConfig) return res.concat(this.getList(groupConfig.list));
+      } else {
+        const btnConfig = this.getButton(btnName);
+        if (btnConfig) res.push(btnConfig);
+      }
+      return res;
+    }, [] as ESLShareButtonConfig[]);
+  }
+
+  /**
+   * Gets the group of buttons configuration.
+   * @returns config of group
+   */
+  public getGroup(groupName: string): ESLShareGroupConfig | undefined {
+    return getConfigSectionItem(this.groups, groupName);
+  }
+
+  /**
+   * Gets the button configuration.
+   * @returns config of button
+   */
+  public getButton(name: string): ESLShareButtonConfig | undefined {
+    return getConfigSectionItem(this.buttons, name);
+  }
+
+  /**
+   * Appends button (inserts or updates) to the current config.
+   * @returns config instance
+   */
+  public appendButton(button: ESLShareButtonConfig): ESLShareConfig {
+    setConfigSectionItem(this.buttons, button);
+    this._onUpdate();
+    return this;
+  }
+
+  /**
+   * Appends group (inserts or updates) to the current config.
+   * @returns config instance
+   */
+  public appendGroup(group: ESLShareGroupConfig): ESLShareConfig {
+    setConfigSectionItem(this.groups, group);
+    this._onUpdate();
+    return this;
+  }
+
+  @decorate(debounce, 25)
+  protected _onUpdate(): void {
+    this.dispatchEvent(new CustomEvent('change'));
+  }
+
+  public override addEventListener(callback: EventListener): void;
+  public override addEventListener(event: 'change', callback: EventListener): void;
+  public override addEventListener(event: any, callback: EventListener = event): void {
+    if (typeof event === 'string' && event !== 'change') {
+      console.warn(`[ESL]: ESLShareConfig does not support '${event}' type`);
+      return;
+    }
+
+    super.addEventListener('change', callback);
+  }
+
+  public override removeEventListener(callback: EventListener): void;
+  public override removeEventListener(event: 'change', callback: EventListener): void;
+  public override removeEventListener(event: any, callback: EventListener = event): void {
+    if (typeof event === 'string' && event !== 'change') return;
+
+    super.removeEventListener('change', callback);
+  }
```

Comment:
Potentially, we can use built-in declaration.
Due to it's mostly inner objects that do not pretend to be observed outside the ESLShare module, we can ignore incorrect subscriptions without special warnings.

Reply:
At first, I had some problems with listeners... The overrides solved the problem. But now it turned out to be unnecessary. And I prefer the option without them.